noir-projects/aztec-nr/aztec/src/context/public_context.nr

```diff
@@ -169,7 +169,9 @@ impl PublicContext {
             historical_header: self.inputs.historical_header,
             prover_address: self.prover_address,
             revert_code: 0,
-            gas_left: self.inputs.call_context.gas_left
+            start_gas_left: self.inputs.gas_left,
+            end_gas_left: self.inputs.gas_left, // AVM should decrease this value
+            transaction_fee: self.inputs.transaction_fee
```

Comment:
Would the public kernel check that previous end gas is the current's start gas to ensure there are no gas gaps?

Reply:
Yep! Need to write something on the list of all kernel checks needed.